packages/react/src/Panels/WorkspacePanel.tsx

```diff
@@ -161,6 +161,7 @@ function EditorSection({ theme, tutorialStore, hasEditor }: PanelProps) {
         onHelpClick={lessonFullyLoaded ? onHelpClick : undefined}
         onFileSelect={(filePath) => tutorialStore.setSelectedFile(filePath)}
         onFileTreeChange={editorConfig.fileTree.allowEdits ? onFileTreeChange : undefined}
```

Comment:
Oh I see! If the callback is not provided then we don't allow any edits. I'm not sure I'm a fan of this approach, my instinct would have been that if `allowEditPatterns` is `undefined` then nothing is allowed as opposed to everything.

I would even go further and say that all those should have the same behaviour:

```ts
allowEditPatterns: [],
allowEditPatterns: undefined,
allowEditPatterns: [!**],
```

Reply:
Yep, it was implemented like this in https://github.com/stackblitz/tutorialkit/pull/314 as we didn't have `allowEditPatterns`. It was simply on/off. Note that `allowEditPatterns` is internal property at this level. TutorialKit authors use `editor.fileTree.allowEdits` only.

So do you mean that in `@tutorialkit/react`, when users use `<FileTree>` directly, we should disable calling given `props.onFileChange` when users don't explicitly pass `props.allowEditPatterns`?

Currently it works like this:

```jsx
// files are not editable
<FileTree />

// any file is editable
<FileTree onFileChange={console.log} />

// specific files are editable
<FileTree onFileChange={console.log} allowEditPatterns={["**.ts"]} />
```

But with that proposed change it could be

```jsx
// files are not editable
<FileTree />

// files are not editable, onFileChange is not called
<FileTree onFileChange={console.log} />

// specific files are editable
<FileTree onFileChange={console.log} allowEditPatterns={["**.ts"]} />
```

Reply:
> So do you mean that in `@tutorialkit/react`, when users use `<FileTree>` directly, we should disable calling given `props.onFileChange` when users don't explicitly pass `props.allowEditPatterns`?

Correct!

In your example, I find the second version to be the least suprising because it matches the same experience that you get with a `<button>` for instance. A button would not be in a disabled state when it doesn't have any callback provided and suddenly become enabled when it has one.

Here it's the same idea (albeit a negated version). Whether `onFileChange` is undefined or not should not impact whether edits are allowed or not. This should be driven entirely by the `allowEditPatterns` property IMO.